code-dot-org/code-dot-org
# Prevent saving lesson edit if database changed

When saving updates to a lesson we check if there is a difference between the initial lesson data for the page and the current lesson data based on the database in order to detect if changes were made on another page such as the Script Edit page which would be overwritten by the lesson update. If we find a difference an error is raised and the user is told to reload the page before saving.

## Links



- [jira](https://codedotorg.atlassian.net/browse/PLAT-484)

## Testing story

Added two tests to lessons_controller_test
- Check that error is raised when there is a diff
- Check that update succeeds when things do match

# Reviewer Checklist:
